### PR TITLE
MULE-1187 Add the concept of namespace and modules in binding context

### PR DESCRIPTION
**Pull request by machaval**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/el/AbstractExpressionModuleBuilderFactory.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  static {
+    try {
+      final AbstractExpressionModuleBuilderFactory factory = load(AbstractExpressionModuleBuilderFactory.class).iterator().next();
+      LOGGER.info(format("Loaded ExpressionModuleBuilderFactory implementation '%s' form classloader '%s'",
+                         factory.getClass().getName(), factory.getClass().getClassLoader().toString()));
+
+      DEFAULT_FACTORY = factory;
+    } catch (Throwable t) {
+      LOGGER.error("Error loading BindingContextBuilderFactory implementation.", t);
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/el/AbstractExpressionModuleBuilderFactory.java]: Update the error message.

File: src/main/java/org/mule/runtime/api/el/ModuleNamespace.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  @Override
+  public String toString() {
+    return stream(elements)
+        .reduce((identity, accumulator) -> identity + "::" + accumulator)
+        .orElse("");
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o)
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/el/ModuleNamespace.java]: @elrodro83, shouldn't this be validated by our formatter?

Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/el/ModuleNamespace.java]: We don't allow if statements without brackets. Please add them.

File: src/main/java/org/mule/runtime/api/el/ModuleNamespace.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      return true;
+    if (o == null || getClass() != o.getClass())
+      return false;
+
+    ModuleNamespace that = (ModuleNamespace) o;
+
+    // Probably incorrect - comparing Object[] arrays with Arrays.equals
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/el/ModuleNamespace.java]: At this point you know its a String[] and this equals will run the String equals. I don't see the issue for this comment. Am I missing something?

File: src/main/java/org/mule/runtime/api/el/ModuleNamespace.java (hunk 2 above)
Comment (machaval) [inline src/main/java/org/mule/runtime/api/el/ModuleNamespace.java]: The build runs perfectly fine [Build Ok](https://ibb.co/dgvgo5)

**Pull request merged**